llvm/test/Verifier/ptrauth-operand-bundles.ll

```diff
@@ -2,30 +2,87 @@
 
 declare void @g()
 
-define void @test_ptrauth_bundle(i64 %arg0, i32 %arg1, ptr %arg2) {
+define void @test_ptrauth_bundle(i64 %arg.64, ptr %arg.ptr, ptr %ok) {
 
-; CHECK: Multiple ptrauth operand bundles
-; CHECK-NEXT: call void %arg2() [ "ptrauth"(i32 42, i64 100), "ptrauth"(i32 42, i64 %arg0) ]
-  call void %arg2() [ "ptrauth"(i32 42, i64 100), "ptrauth"(i32 42, i64 %arg0) ]
+; CHECK: Multiple ptrauth operand bundles on a function call
+; CHECK-NEXT: call void %arg.ptr() [ "ptrauth"(i64 42, i64 100), "ptrauth"(i64 42, i64 %arg.64) ]
+  call void %arg.ptr() [ "ptrauth"(i64 42, i64 100), "ptrauth"(i64 42, i64 %arg.64) ]
 
-; CHECK: Ptrauth bundle key operand must be an i32 constant
-; CHECK-NEXT: call void %arg2() [ "ptrauth"(i32 %arg1, i64 120) ]
-  call void %arg2() [ "ptrauth"(i32 %arg1, i64 120) ]
+; CHECK: Direct call cannot have a ptrauth bundle
+; CHECK-NEXT: call void @g() [ "ptrauth"(i64 42, i64 120) ]
+  call void @g() [ "ptrauth"(i64 42, i64 120) ]
 
-; CHECK: Ptrauth bundle key operand must be an i32 constant
-; CHECK-NEXT: call void %arg2() [ "ptrauth"(i64 42, i64 120) ]
-  call void %arg2() [ "ptrauth"(i64 42, i64 120) ]
+; CHECK-NOT: call void %ok()
+  call void %ok() [ "ptrauth"(i32 42, i64 120) ]   ; OK
+  call void %ok() [ "ptrauth"(i32 42, i64 %arg.64) ] ; OK
+  call void %ok() [ "ptrauth"(i64 %arg.64, i64 123) ] ; OK
+  call void %ok() [ "ptrauth"(i64 %arg.64, i64 123, i64 %arg.64, i64 42) ] ; OK
```

Comment:
BTW, what behavior do we expect from further lowering of these? For example, if we run `llc` for aarch64-linux-gnu with pauth CPU feature enabled on smth like the following, what is the expected output?

I suppose that the bundle should be treated as unexpected during lowering, but I fail to find the test ensuring that. Could you please refer me to that test?

```
define void @test(i64 %arg.64, ptr %ok) {
  call void %ok() [ "ptrauth"(i64 %arg.64, i64 123, i64 %arg.64, i64 42) ]
}
```

Reply:
You are right, this particular case is not intended to be a valid code on AArch64, it is only valid "in general", as a target-neutral code.

Having test cases on unsupported ptrauth bundles being rejected by the backend seems reasonable, thanks!